### PR TITLE
fix: [DHIS2-18019] related stages UI tweaks

### PR DESCRIPTION
**Pull request by simonadomnisoru**
[DHIS2-18019](https://dhis2.atlassian.net/browse/DHIS2-18019)

**Tech summary:**
- moved the back button and the widgets actions buttons
- remove the WidgetTwoEventWorkspace from the default page layout and move it inside WidgetEditEvent
- added type property to WidgetTwoEventWorkspace. If enabled via a custom page layout, the WidgetTwoEventWorkspace can still be used and moved around as a standalone widget.

[DHIS2-17019]: https://dhis2.atlassian.net/browse/DHIS2-17019?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DHIS2-18019]: https://dhis2.atlassian.net/browse/DHIS2-18019?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    icon: {
+        marginRight: spacersNum.dp8,
+    },
+    decription: {
+        margin: `${spacersNum.dp8}px 0`,
+    },
+};
+
+const WidgetWrapperPlain = ({ widget, type, classes }: Props) => {
+    if (type === WidgetTwoEventWorkspaceWrapperTypes.EDIT_EVENT) {
```
Comment (eirikhaugstulen) [inline src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js:43]: Are there any plans of having multiple TwoEventWorkspaces? Is this to support a standalone widget in the future?

File: src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        return (
+            <div className={classes.container}>
+                <div className={classes.header}> {i18n.t('Refferal response')} </div>
```
Comment (eirikhaugstulen) [inline src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js:46]: We've always tried to stay away from using the word referral as there are other potential use-cases for linked stages. Should we switch this to a more generic term?

Edit: based on the design, this actually looks like the stage name of the stage we are currently in. We are adding data to the referral response stage.

File: src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                <div className={classes.referalResponse}>
+                    <div className={classes.linkedEvent}>
+                        <span className={classes.icon}>
+                            <IconLink16 />
+                        </span>
+                        <div>{i18n.t('Linked event')}</div>
+                    </div>
+                    <div className={classes.decription}>
+                        {i18n.t(
+                            'This refferal response event is linked to a Refferal details event. Review the linked event details before entering data below',
```
Comment (eirikhaugstulen) [inline src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js:56]: Same here. Should we consider using the stage name instead? 

`This refferal response event is linked to a {{linkedStageName}} event.`

File: src/core_modules/capture-core/components/Pages/EnrollmentEditEvent/PageLayout/DefaultPageLayout.constants.js [hunk 4]
```diff
@@ -22,10 +22,6 @@ export const WidgetsForEnrollmentEventEdit: $ReadOnly<{ [key: string]: WidgetCon
 
 export const DefaultPageLayout: PageLayoutConfig = {
     leftColumn: [
-        {
-            type: WidgetTypes.COMPONENT,
-            name: 'TwoEventWorkspace',
-        },
```
Comment (eirikhaugstulen) [inline src/core_modules/capture-core/components/Pages/EnrollmentEditEvent/PageLayout/DefaultPageLayout.constants.js:28]: I think we need to remove "TwoEventWorkspace" from the available components as well: `src/core_modules/capture-core/components/Pages/common/EnrollmentOverviewDomain/EnrollmentPageLayout/DefaultEnrollmentLayout.types.js` as well.

**Review by eirikhaugstulen — COMMENTED** [review]
Hey @simonadomnisoru! I think most of the code looks solid here, but I have some questions about the design. Let's either discuss this internally or take it up with the design team. Other than that; very good work! 🎉

**Comment by eirikhaugstulen** [comment]
Also, now that we're no longer bound to allowing users to define where they want this widget to be themselves, should we implement the over/under functionality the design team wanted? Open for suggestions. @simonadomnisoru @JoakimSM

File: src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js (hunk 1 above)
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js:43]: Yes the `type` logic is only to support a standalone widget

**Review by eirikhaugstulen — APPROVED** [review]
Looks like a great upgrade, @simonadomnisoru! Awaiting response from Joe, but it looks good from my perspective now! 🎉

File: src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        fontSize: 16,
+        color: colors.grey800,
+    },
+    referalResponse: {
+        padding: spacersNum.dp16,
+        backgroundColor: colors.blue100,
+        borderRadius: '3px',
+    },
+    linkedEvent: {
+        color: colors.blue600,
```
Comment (cooper-joe) [inline src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js:29]:
```suggestion
        color: colors.blue900,
```

File: src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+const WidgetWrapperPlain = ({ widget, type, stage, linkedStage, classes }: Props) => {
+    if (type === WidgetTwoEventWorkspaceWrapperTypes.EDIT_EVENT) {
+        return (
+            <div className={classes.container}>
+                <div className={classes.header}> {stage?.name} </div>
+                <div className={classes.referalResponse}>
+                    <div className={classes.linkedEvent}>
+                        <span className={classes.icon}>
+                            <IconLink16 />
```
Comment (cooper-joe) [inline src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js:51]: Icon color should be `blue800`.

**Review by cooper-joe — APPROVED** [review]
LGTM, approved with minor changes.

File: src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js (hunk 5 above)
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetWrapper/WidgetWrapper.container.js:29]: fix: text color

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.42 version

**Pull request merged**